packages/blockchain-wallet-v4/src/types/HDAccount.js

```diff
@@ -1,27 +1,41 @@
+import * as Cache from './Cache'
+import * as crypto from '../walletCrypto'
+import * as Derivation from './Derivation'
+import * as DerivationList from './DerivationList'
 import {
   assoc,
   compose,
+  contains,
   curry,
   dissoc,
-  equals,
   is,
-  isEmpty,
   isNil,
   not,
   pipe,
-  split,
-  values
+  split
 } from 'ramda'
-import { over, traverseOf, view } from 'ramda-lens'
-import Bitcoin from 'bitcoinjs-lib'
-import Task from 'data.task'
 /* eslint-disable */
-import * as AddressLabelMap from './AddressLabelMap' // if we delete this import, wallet tests will fail -  ¯\_(ツ)_/¯
-import * as Cache from './Cache'
-import * as crypto from '../walletCrypto'
-import { fromJS as iFromJS } from 'immutable-ext'
+import { fromJS as iFromJS } from 'immutable-ext' // if we delete this import, wallet tests will fail -  ¯\_(ツ)_/¯
+/* eslint-disable */
+import { over, traversed, traverseOf, view } from 'ramda-lens'
+import Task from 'data.task'
 import Type from './Type'
-/* eslint-enable */
+
+export const DEFAULT_DERIVATION_TYPE = 'segwitP2SH'
```

Comment:
bitcoin core now uses native segwit as default, this should be the same I think... default should probably be p2wpkh / p2wsh

Reply:
Thank you for having a look at the PR. Native segwit will come after this PR.

The majority of the work in this PR is to support multiple derivation types per account. Once we have segwitP2SH support adding native segwit becomes much easier. We could support native segwit as the default deriviation type one day, but there are still some services that don't support sending to native segwit, that is why it should be opt in for now.

Reply:
do you mean spending from native segwit (bech32)? which services are linked to the wallet that don't support? I guess bip84 would be then used instead of 44 in that PR.. maybe if those services don't support I can take a look and perhaps get in touch to see when they will support or help out if its open source.

Reply:
I mean sending to native segwit, some services don't support that. Only giving users a native segwit address to receive from would lock them into only using services that support sending to native segwit. Binance, for example, does not support sending to native segwit.

Reply:
Ah I see so you have to use legacy by default for now. Got it.

Reply:
also one more thing, if your supporting bip49 then will api work with ypub?

Reply:
Maybe descriptors should be used instead to avoid xpub ypub zpub hell: https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md

Reply:
The API (multiaddr and unspent) will accept an optional parameter activeP2SH for the xpubs derived from the segwitP2SH chain (49)

Reply:
Cool maybe it can be like ?bip49=xpub3|xpub4 so when bip84 comes it will be another extension.